### PR TITLE
feature gates for collectors

### PR DESCRIPTION
**Pull request by piontec**
This PR adds:
- `enable*Receiver` flags to enable configuration of different incoming protocols (collectors)
- templated port numbers for collectors (to avoid typos)
- configurable upstream Jaeger server URL

### CODE REVIEW

File: helm/opencensus-collector-app/templates/configmap.yaml [hunk 1]
```diff
@@ -8,21 +8,29 @@ metadata:
...
           server-parameters:
             max-connection-age: 120s
-            max-connection-age-grace: 30s            
+            max-connection-age-grace: 30s
+{{- end }}
+{{- if .Values.enableZipkinReceiver }}
       zipkin:
-        port: 9411
-      jaeger: {}
+        port: {{ .Values.ports.zipkinReceiver }}
```
Comment (cokiengchiara) [inline helm/opencensus-collector-app/templates/configmap.yaml:21]: just curious: Why don't you name `port` more specifically similar to `jaeger-thrift-tchannel-port`, e.g. `zipkin-port`?

File: helm/opencensus-collector-app/templates/deployment.yaml [hunk 2]
```diff
@@ -29,18 +29,17 @@ spec:
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         resources:
           {{- toYaml .Values.resources | nindent 12 }}
```
Comment (cokiengchiara) [inline helm/opencensus-collector-app/templates/deployment.yaml:31]: What is `- toYaml` for? Shouldn't it be just `.Values.resources`?

File: helm/opencensus-collector-app/values.yaml [hunk 3]
```diff
@@ -23,3 +23,15 @@ resources:
   requests:
     cpu: 200m
     memory: 512Mi
+
+enableOpencensusReceiver: true
+enableZipkinReceiver: true
+enableJaegerReceiver: true
```
Comment (cokiengchiara) [inline helm/opencensus-collector-app/values.yaml:29]: What do these do?

**Review by cokiengchiara — APPROVED** [review]
LGTM if you tested it haha. Had some questions for learning :)

File: helm/opencensus-collector-app/templates/configmap.yaml (hunk 1 above)
Comment (piontec) [reply helm/opencensus-collector-app/templates/configmap.yaml:21]: Just to make the config more structured. You could do this differently, ie. organize everything around different ingest protocols.

File: helm/opencensus-collector-app/templates/deployment.yaml (hunk 2 above)
Comment (piontec) [reply helm/opencensus-collector-app/templates/deployment.yaml:31]: Normally, Values are parsed in to go data structures. To put them back in YAML, you need `toYaml`

File: helm/opencensus-collector-app/values.yaml (hunk 3 above)
Comment (piontec) [reply helm/opencensus-collector-app/values.yaml:29]: As part of the exercise: read the changed templates and try to figure out what they do :) If you're lost, we can have a call about this. They are meant to allow the user to configure receivers.

Comment (cokiengchiara) [reply helm/opencensus-collector-app/values.yaml:29]: thanks! "read the changed templates and try to figure out what they do" yes i tried to do that! For this case, in the original template, I didn't see these parameters (or whatever you call the `enableOpencensusReceiver:` thingies. Got it now! So you can have `enableRandomReceiver: false`

Comment (piontec) [reply helm/opencensus-collector-app/values.yaml:29]: Exactly. Application's user can choose this way which receivers should we enable and our chart will include conditionally the necessary config stuff.

**Pull request merged**